flocker/control/schema/types.yml

```diff
@@ -59,7 +59,17 @@ definitions:
         external:
           type: integer
     uniqueItems: true
-    
+
+  environment:
+    title: "Container environment variables"
+    description: "Mapping of label and value pairs for environment variables"
+    type: object
+    patternProperties:
+      "^.{1,4096}$":
+        type: string
+        maxLength: 4096
+    additionalProperties: false
```

Comment:
This schema doesn't limit the number of properties on the object - so the total size of the environment is not limited.  Limiting the total size of the environment seems like the primary motivation to have any limits at all in the schema here.